crates/robonode-server/src/logic/op_authenticate.rs

```diff
@@ -140,39 +146,47 @@ where
             .await;
 
         let results = match db_search_result_adapter(search_result) {
-            DbSearchResult::OtherError(err) => return Err(Error::InternalErrorDbSearch(err)),
+            DbSearchResult::OtherError(err) => {
+                return Err(Error::InternalErrorDbSearch(err, scan_result_blob))
+            }
             DbSearchResult::NoGroupError => {
                 trace!(message = "Got no-group error instead of FaceTec 3D-DB search results, assuming no results");
                 vec![]
             }
             DbSearchResult::Response(search_res) => {
                 trace!(message = "Got FaceTec 3D-DB search results", ?search_res);
                 if !search_res.success {
-                    return Err(Error::InternalErrorDbSearchUnsuccessful);
+                    return Err(Error::InternalErrorDbSearchUnsuccessful(scan_result_blob));
                 }
                 search_res.results
             }
         };
 
         // If the results set is empty - this means that this person was not
         // found in the system.
-        let found = results.first().ok_or(Error::PersonNotFound)?;
+        let found = results
+            .first()
+            .ok_or(Error::PersonNotFound(scan_result_blob.clone()))?;
         if found.match_level < MATCH_LEVEL {
-            return Err(Error::InternalErrorDbSearchMatchLevelMismatch);
+            return Err(Error::InternalErrorDbSearchMatchLevelMismatch(
+                scan_result_blob,
+            ));
         }
 
-        let public_key_bytes =
-            hex::decode(&found.identifier).map_err(|_| Error::InternalErrorInvalidPublicKeyHex)?;
-        let public_key =
-            PK::try_from(&public_key_bytes).map_err(|_| Error::InternalErrorInvalidPublicKey)?;
+        let public_key_bytes = hex::decode(&found.identifier)
+            .map_err(|_| Error::InternalErrorInvalidPublicKeyHex(scan_result_blob.clone()))?;
```

Comment:
Use `match` to avoid a clone here and in similar locations

Reply:
https://github.com/humanode-network/humanode/pull/1065/commits/461f645870a1df236da551e5a596424153e99a6e